762245453 modules cms content extra features

orthener opened the pull request:
https://github.com/Overdose-Digital/magento2-cms-management/issues/10

- separate import/export
Overdose/CMSContent/Model/Import.php
Overdose/CMSContent/Model/Export.php
Overdose/CMSContent/Api/ContentImportInterface.php
Overdose/CMSContent/Api/ContentExportInterface.php
- add export to xml
- add export each CMS entity to separate files in one archive
- add import to xml
- fix import to json
- refactoring (adjusted the code for PHP 7.3 version)
- added bulk import from config folder (by cli or s:up)
- fixed few issues with pages import/export (cases with pages that contain blocks)

AntonBalagurovOD committed:
Add export to XML

AntonBalagurovOD committed:
Export to separate files

AntonBalagurovOD committed:
Separate import and export

AntonBalagurovOD committed:
Add import from xml

Commit:
added split export/import functionality - https://www.wrike.com/open.htm?id=762245453

Commit:
added bulk import functionality from config - https://www.wrike.com/open.htm?id=762245453

Commit:
customized export from admin panel to special XML format - https://www.wrike.com/open.htm?id=762245453

Commit:
customized import from admin panel to special XML format, fixes, refactoring - https://www.wrike.com/open.htm?id=762245453

Commit:
added clearing the content version table - https://www.wrike.com/open.htm?id=762245453

Commit:
added multi-store support to XML export/import - https://www.wrike.com/open.htm?id=762245453

Commit:
refactoring - https://www.wrike.com/open.htm?id=762245453

Commit:
removed and refactoring store mapping functionality - https://www.wrike.com/open.htm?id=762245453

Commit:
added own logger and error handling functionality - https://www.wrike.com/open.htm?id=762245453

Commit:
moved sample files to od_cms folder - https://www.wrike.com/open.htm?id=762245453

vitiodev committed:
Added cron for delete old files

vitiodev committed:
Update readme.md

Commit:
replaced deprecated model saving by repository - https://www.wrike.com/open.htm?id=762245453

Commit:
added duplication import files records handler - https://www.wrike.com/open.htm?id=762245453

Commit:
refactoring - https://www.wrike.com/open.htm?id=762245453

Commit:
added new sample file for the split mode - https://www.wrike.com/open.htm?id=762245453

Commit:
refactoring (renamed Json to JSON) - https://www.wrike.com/open.htm?id=762245453

Commit:
fixed issue with incorrect multi-store saving - https://www.wrike.com/open.htm?id=762245453

Commit:
updated README.md accordingly to provided customizations - https://www.wrike.com/open.htm?id=762245453

onlinebizsoft commented:
@orthener great to see :)

KSergiy commented on Api/CmsEntityConverterInterface.php:
```diff
@@ -0,0 +1,26 @@
+<?php
```
Need to use strict_type declaration in there and in the next files.

declare(strict_types=1);

KSergiy commented on Api/CmsEntityConverterInterface.php:
```diff
@@ -0,0 +1,26 @@
+    const BLOCK_ENTITY_CODE = 'blocks';
+
+    /**
+     * @return string
```
Need to describe what this function must do ( correct description must be everywhere )

KSergiy commented on Api/CmsEntityConverterInterface.php:
```diff
@@ -0,0 +1,26 @@
+
+    /**
+     * @param array $cmsEntities
+     * @return array
```
According to standards between: `@return` and `@param` must be one blank line:
`
* @param something
*
* @return something
`

onlinebizsoft commented:
Nice progress!

SerhiiKoblia committed:
[d-b-f] Updated:

- refactored code

KSergiy committed:
[842741420] Updated:

- getting history directory duging creating history backups
- clearing backups

Task: https://www.wrike.com/open.htm?id=842741420

KSergiy committed:
[762245453] Updated:

- refactored code
- changed flow of getting convertor and generator

Task: https://www.wrike.com/open.htm?id=762245453

KSergiy committed:
[762245453] Updated:

- refactored code
- optimized converted modules and actions

Task: https://www.wrike.com/open.htm?id=762245453

KSergiy committed:
Merge branch 'delete-backup-files' into 762245453-modules-cms-content-extra-features

KSergiy committed:
[762245453] Updated:

- changed backups folders
- refactored position of content objects
- deleted some classes

Task: https://www.wrike.com/open.htm?id=762245453

KSergiy committed:
[762245453] Updated:

- updated showing history on entites
- updated flow of deleting an old items
- refactore part of code

Task: https://www.wrike.com/open.htm?id=762245453

KSergiy committed:
[762245453] Updated:

- made code refactoring during testing code

Task: https://www.wrike.com/open.htm?id=762245453

KSergiy committed:
[762245453] Updated:

- added fixes
- refactored system configs

Task: https://www.wrike.com/open.htm?id=762245453

KSergiy committed:
[762245453] Updated:

- import form was totaly refactored
- import actions was refactored
- readme data

Task: https://www.wrike.com/open.htm?id=762245453

KSergiy committed:
[762245453] Updated:

- added Unit tests
- fixed small issues
- changed module version

Task: https://www.wrike.com/open.htm?id=762245453

orthener committed:
beautify XML

orthener committed:
Update changelog
